internal/core/textlayout/sharedparley.rs

```diff
@@ -0,0 +1,138 @@
+        if let Some(family) = &font_request.family {
+            builder.push_default(parley::StyleProperty::FontStack(
+                parley::style::FontStack::Single(parley::style::FontFamily::Named(
+                    family.as_str().into(),
+                )),
+            ));
+        }
+        if let Some(weight) = font_request.weight {
+            builder.push_default(parley::StyleProperty::FontWeight(
+                parley::style::FontWeight::new(weight as f32),
+            ));
+        }
+        if let Some(letter_spacing) = font_request.letter_spacing {
+            builder.push_default(parley::StyleProperty::LetterSpacing(letter_spacing.get()));
+        }
+        builder.push_default(parley::StyleProperty::FontStyle(if font_request.italic {
+            parley::style::FontStyle::Italic
+        } else {
+            parley::style::FontStyle::Normal
+        }));
+    }
+    let pixel_size = options
+        .font_request
+        .and_then(|font_request| font_request.pixel_size)
+        .unwrap_or(DEFAULT_FONT_SIZE);
+    builder.push_default(parley::StyleProperty::FontSize(pixel_size.get()));
+    builder.push_default(parley::StyleProperty::WordBreak(match options.text_wrap {
+        TextWrap::NoWrap => parley::style::WordBreakStrength::KeepAll,
+        TextWrap::WordWrap => parley::style::WordBreakStrength::Normal,
+        TextWrap::CharWrap => parley::style::WordBreakStrength::BreakAll,
+    }));
+    if options.text_overflow == TextOverflow::Elide {
+        todo!();
+    }
+
+    builder.push_default(parley::StyleProperty::Brush(Brush {
+        stroke: options.stroke,
+        ..Default::default()
+    }));
+    if let Some(selection) = options.selection {
+        builder.push(
+            parley::StyleProperty::Brush(Brush { stroke: options.stroke, is_selected: true }),
+            selection,
+        );
+    }
+
+    let mut layout: parley::Layout<Brush> = builder.build(text);
+    layout.break_all_lines(options.max_width.map(|max_width| max_width.get() * scale_factor));
+    layout.align(
+        options.max_width.map(|max_width| max_width.get() * scale_factor),
+        match options.horizontal_align {
+            TextHorizontalAlignment::Left => parley::Alignment::Left,
+            TextHorizontalAlignment::Center => parley::Alignment::Middle,
+            TextHorizontalAlignment::Right => parley::Alignment::Right,
+        },
+        parley::AlignmentOptions::default(),
+    );
+
+    let y_offset = match (options.max_height, options.vertical_align) {
+        (Some(max_height), TextVerticalAlignment::Center) => {
+            (max_height.get() - layout.height()) / 2.0
+        }
+        (Some(max_height), TextVerticalAlignment::Bottom) => max_height.get() - layout.height(),
+        (None, _) | (Some(_), TextVerticalAlignment::Top) => 0.0,
+    };
+
+    Layout { inner: layout, y_offset }
+}
+
+pub struct Layout {
+    pub inner: parley::Layout<Brush>,
```

Comment:
Since the user of this structure is primarily going to operate on `inner`, perhaps `inner` should receive a more pronounced name :). Or, alternatively, `Layout` could deref to `parley::Layout` - like we do for the font collection.